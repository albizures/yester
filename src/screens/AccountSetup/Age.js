import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View, StyleSheet } from 'react-native'

import Trasnlate from '../../components/Translate'

export default class AgeSetup extends Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
  }
  render () {
    const { name } = this.props
    return (
      <View style={styles.container}>
        <View style={styles.top}>
          <Trasnlate style={styles.text} keyName='setup.age.greeting' data={{ name }} />
          <Trasnlate style={styles.text} keyName='setup.age.greeting.subtitle' data={{ name }} />
        </View>
        <Trasnlate style={styles.text} keyName='setup.age.question' data={{ name }} />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  text: {
    textAlign: 'center',
  },
  top: {
    marginHorizontal: 50,
  },
  container: {
    flex: 1,
    alignItems: 'center',
    textAlign: 'center',
    margin: 10,
  },
})
