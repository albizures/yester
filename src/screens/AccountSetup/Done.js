import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'

import Trasnlate from '../../components/Translate'

export default class Done extends Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
    place: PropTypes.string.isRequired,
  }
  render () {
    const { name, place } = this.props
    return (
      <View>
        <Trasnlate keyName='setup.done.title' data={{ name, place }} />
        <Trasnlate keyName='setup.done.subtitle' />
      </View>
    )
  }
}
