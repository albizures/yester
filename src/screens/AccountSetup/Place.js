import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'

import Trasnlate from '../../components/Translate'

export default class Done extends Component {
  static propTypes = {
    year: PropTypes.number.isRequired,
  }
  render () {
    const { year } = this.props
    return (
      <View>
        <Trasnlate keyName='setup.place.title' data={{ year }} />
        <Trasnlate keyName='setup.place.subtitle' />
        <Trasnlate keyName='setup.place.form.title' />
      </View>
    )
  }
}
