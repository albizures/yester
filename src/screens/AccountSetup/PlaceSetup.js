import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View, Text } from 'react-native'

// import Trasnlate from '../../components/Translate'

export default class Done extends Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
  }
  render () {
    const { name } = this.props
    return (
      <View>
        <Text>place {name}</Text>
      </View>
    )
  }
}
