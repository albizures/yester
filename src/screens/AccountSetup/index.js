import React, { Component } from 'react'
import { Button, StyleSheet, View } from 'react-native'

import Container from '../../components/Container'
import Stepper from '../../components/Stepper'

import Age from './Age'
import Done from './Done'
import Place from './Place'

import colors from '../../utils/colors'

export default class AccountSetup extends Component {
  stepper = React.createRef()
  state = {}

  getBottomBar = (options) => {
    const { next } = options

    return (
      <View style={styles.footer}>
        <Button style={styles.button} title='Continue' onPress={next} />
      </View>
    )
  }

  render () {
    return (
      <Container>
        <Stepper ref={this.stepper} scrollEnabled={false} bottomBar={this.getBottomBar}>
          {[
            <Age key={0} name='John' />,
            <Place key={1} year={60} />,
            <Done key={2} name='John' place={'Roseville - California'} />,
          ]}
        </Stepper>
      </Container>
    )
  }
}

const styles = StyleSheet.create({
  button: {
    height: 30,
    backgroundColor: '#f00',
  },
  footer: {
    height: 50,
    width: '100%',
    padding: 10,
    backgroundColor: colors.gray,
  },
})
