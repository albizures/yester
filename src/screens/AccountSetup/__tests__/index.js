import React from 'react'
import renderer from 'react-test-renderer'

import AccountSetup from '../'

const navigation = {
  navigate: jest.fn(),
}

describe('src/screens/AccountSetup', () => {
  it('should render', () => {
    const tree = renderer.create(<AccountSetup navigation={navigation} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
