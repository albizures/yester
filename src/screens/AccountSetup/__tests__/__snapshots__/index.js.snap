// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/screens/AccountSetup should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          <AgeSetup
            name="John"
          />,
          <Done
            year={60}
          />,
          <Done
            name="John"
            place="Roseville - California"
          />,
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onViewableItemsChanged={[Function]}
      pagingEnabled={true}
      renderItem={[Function]}
      scrollEnabled={false}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={
        Array [
          Object {
            "onViewableItemsChanged": [Function],
            "viewabilityConfig": undefined,
          },
        ]
      }
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "column",
                },
                Object {
                  "height": 1334,
                  "justifyContent": "center",
                  "paddingTop": 0,
                  "width": 750,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "margin": 10,
                  "textAlign": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "marginHorizontal": 50,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  Hi John üëã
                </Text>
                <Text
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  That's a great name! Now, let's meet properly to get you started
                </Text>
              </View>
              <Text
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                When di you saw this beautiful Earth for the first time?
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "column",
                },
                Object {
                  "height": 1334,
                  "justifyContent": "center",
                  "paddingTop": 0,
                  "width": 750,
                },
              ]
            }
          >
            <View>
              <Text>
                The 60's! Awesome! üôå
              </Text>
              <Text>
                What a great era for this legend to begin.
              </Text>
              <Text>
                Where in the world did you happen to born?
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "column",
                },
                Object {
                  "height": 1334,
                  "justifyContent": "center",
                  "paddingTop": 0,
                  "width": 750,
                },
              ]
            }
          >
            <View>
              <Text>
                Alrighty, John from Roseville - California, Nice to meet you ü§ù
              </Text>
              <Text>
                This is where we start our journey of writing your incredible story
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "backgroundColor": "#c3c3c3",
          "height": 50,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Continue
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
